tldw_Server_API/app/api/v1/endpoints/media.py

```
@@ -276,6 +276,40 @@
+        try:
+            ip = ipaddress.ip_address(host)
+            # Disallow internal/private IP ranges
+            if ip.is_private or ip.is_loopback or ip.is_reserved or ip.is_link_local:
+                return False
+        except ValueError:
+            # Not an IP address; allow domain unless blacklisted
+            pass
+        # Disallow specific cloud metadata endpoint
+        if host == "169.254.169.254":
+            return False
+        return True
```

Ruff (0.14.3)

Consider moving this statement to an `else` block

(TRY300)